dot/core/messages.go

```diff
@@ -17,29 +17,31 @@
 package core
 
 import (
+	"reflect"
+
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/transaction"
 )
 
 // HandleTransactionMessage validates each transaction in the message and
 // adds valid transactions to the transaction queue of the BABE session
-func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) error {
+func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) (bool, error) {
 	logger.Debug("received TransactionMessage")
 	if !s.isBlockProducer {
-		return nil
+		return false, nil
 	}
 
 	// get transactions from message extrinsics
 	txs := msg.Extrinsics
-
+	var toRemove []types.Extrinsic
 	for _, tx := range txs {
 		// validate each transaction
 		externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
 		val, err := s.rt.ValidateTransaction(externalExt)
 		if err != nil {
 			logger.Error("failed to validate transaction", "err", err)
-			return err
+			return false, err
```

Comment:
don't return here but continue, and mark this tx "do not propagate" instead

Reply:
makes sense, updated.